### PR TITLE
Read structured dataset from a folder 

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
`Hugginface` and `polars` transformer failed to read parquet file generated from other transformers, because those transformers always read the data from a specific path (.../00000), but we may have different file name in that folder.

We should read the data by default from a folder.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 ^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3189

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/flytekit-huggingface/setup.py [hunk 1]
```diff
@@ -38,4 +38,5 @@
         "Topic :: Software Development :: Libraries",
         "Topic :: Software Development :: Libraries :: Python Modules",
     ],
+    entry_points={"flytekit.plugins": [f"{PLUGIN_NAME}=flytekitplugins.{PLUGIN_NAME}"]},
```
Comment (wild-endeavor) [inline plugins/flytekit-huggingface/setup.py:41]: do we want to autoload these?

File: plugins/flytekit-spark/setup.py [hunk 2]
```diff
@@ -34,4 +34,5 @@
         "Topic :: Software Development :: Libraries :: Python Modules",
     ],
     scripts=["scripts/flytekit_install_spark3.sh"],
+    entry_points={"flytekit.plugins": [f"{PLUGIN_NAME}=flytekitplugins.{PLUGIN_NAME}"]},
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply plugins/flytekit-spark/setup.py:37]: I added this because flytekit will auto load `spark` and `huggingface` structured dataset transformer. otherwise, we have to manually import transformer.

we've added the `entry_points` in vaex and polars plugin.

**Review by wild-endeavor — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
